### PR TITLE
error in data inserting

### PR DESCRIPTION
**Pull request by kkaajjaall**
error in data inserting after submit click.

### CODE REVIEW

File: app/Http/Controllers/TestData.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            'employer_name'=>'required',
+            'employer_address'=>'required|min:10|max:100' ,
+            'job_title'=>'required',
+            'employment_status'=>'required',
+            'employment_start'=>'required|date',
+            'employment_end'=>'required|date|after_or_equal:employment_start',
+
+        ]);
+
+        return $request;
```
Comment (amolswnz) [inline app/Http/Controllers/TestData.php:23]: Remove this line

File: app/Http/Controllers/AddressData.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+{
+    public function store(Request $request)
+    {
+        $request->validate([
+
+
+            'permanent_address'=>'required|min:10|max:100' ,
+            'permanent_apartment'=>'required' ,
+            'permanent_city'=>'required' ,
+            'permanent_pincode'=>'required|size:6|digits:6',
```
Comment (ithqindia) [inline app/Http/Controllers/AddressData.php:18]: remove size requirement

File: app/Http/Controllers/AddressData.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+        ]);
+        EmployeAddress::create([
+            'permanent_address' => $request->get('permanent_address'),
+            'permanent_apartment' => $request->get('permanent_apartment'),
+            'permanent_city' => $request->get('permanent_city'),
+            'permanent_pincode' => $request->get('permanent_pincode'),
+            'permanent_state' => $request->get('permanent_state'),
+            'permanent_country' => $request->get('permanent_country'),
+            'permanent_emergancy_contact_name' => $request->get('permanent_emergancy_contact_name'),
```
Comment (ithqindia) [inline app/Http/Controllers/AddressData.php:35]: correct spelling emergency

File: app/Http/Controllers/EmployeData.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    public function store(Request $request)
+    {
+        $request->validate([
+
+            'employer_name' => 'required',
+            'employer_address' => 'required|min:10|max:100',
+            'job_title' => 'required',
+            'employment_status' => 'required',
+            'employment_start' => 'required|date',
+            'employment_end' => 'required|date|after_or_equal:employment_start',
```
Comment (ithqindia) [inline app/Http/Controllers/EmployeData.php:19]: Very good.

File: app/Http/Controllers/LocalAddress.php [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+
+
+class LocalAddress extends Controller
+{
+    public function store(Request $request)
+    {
+        $request->validate([
+
+            'local_add'=>'required|min:10|max:100' ,
```
Comment (ithqindia) [inline app/Http/Controllers/LocalAddress.php:19]: local_address

File: app/Http/Controllers/LocalAddress.php [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+            'local_apartment'=>'required' ,
+            'local_city'=>'required' ,
+            'local_pincode'=>'required|size:6|digits:6',
```
Comment (ithqindia) [inline app/Http/Controllers/LocalAddress.php:22]: Remove size

File: app/Http/Controllers/LocalAddress.php [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+            'local_state'=>'required' ,
+            'local_country'=>'required' ,
+            'local_emergancy_contact_name'=>'required',
```
Comment (ithqindia) [inline app/Http/Controllers/LocalAddress.php:25]: emergency

File: app/Http/Controllers/RefreeData.php [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+class RefreeData extends Controller
+{
+    public function store(Request $request)
+    {
+        $request->validate([
+
+            'first_name' => 'required',
+            'last_name' => 'required',
+            'organization' => 'required',
```
Comment (ithqindia) [inline app/Http/Controllers/RefreeData.php:16]: organisation - British spelling

File: app/Http/Controllers/LocalAddress.php [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+            'local_emergancy_contact_number'=>'required',
+            'local_emergancy_relationship'=>'required',
+
+        ]);
+
+        Data::create([
```
Comment (ithqindia) [inline app/Http/Controllers/LocalAddress.php:31]: Change to LocalData

File: app/Models/Data.php [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Data extends Model
+{
+    use HasFactory;
+    protected $table = "local_addresses_details";
```
Comment (ithqindia) [inline app/Models/Data.php:11]: local_address_details

File: app/Models/Data.php [hunk 11]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Data extends Model
```
Comment (ithqindia) [inline app/Models/Data.php:8]: LocalData

File: app/Models/EmployeAddress.php [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class EmployeAddress extends Model
+{
+    use HasFactory;
+    protected $table = "employe_addresses";
```
Comment (ithqindia) [inline app/Models/EmployeAddress.php:11]: employee_addresses

File: database/migrations/2021_12_13_052856_create_local__addresses_table.php [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+class CreateLocalAddressesTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('local_addresses_details', function (Blueprint $table) {
```
Comment (ithqindia) [inline database/migrations/2021_12_13_052856_create_local__addresses_table.php:16]: local_address_details

File: database/migrations/2021_12_13_052856_create_local__addresses_table.php [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+            $table->id();
+            $table->string('local_add');
```
Comment (ithqindia) [inline database/migrations/2021_12_13_052856_create_local__addresses_table.php:18]: local_address

File: routes/web.php [hunk 15]
```diff
@@ -20,3 +25,21 @@
...
+Route::get('/employe_information', function () {
+    return view('employe_information');
+});
+Route::get('/referee', function () {
+    return view('referee');
+});
+
+
+Route::post('/test1', [EmployeData::class, 'store']);
+Route::post('/test2', [RefreeData::class, 'store']);
```
Comment (ithqindia) [inline routes/web.php:42]: /referee-data

File: routes/web.php [hunk 16]
```diff
@@ -20,3 +25,21 @@
...
+
+Route::post('/test3', [AddressData::class, 'store']);
```
Comment (ithqindia) [inline routes/web.php:44]: /address-data

**Review by ithqindia — CHANGES_REQUESTED** [review]

**Comment by ithqindia** [comment]
Please format all your files and remove empty lines keep it to minimum one empty line.

**Comment by ithqindia** [comment]
app/Http/Controllers/Exceptions/Handler.php why this file is renamed?

File: app/Http/Controllers/AddressData.php [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+            'permanent_address' => $request->get('permanent_address'),
+            'permanent_apartment' => $request->get('permanent_apartment'),
+            'permanent_city' => $request->get('permanent_city'),
+            'permanent_pincode' => $request->get('permanent_pincode'),
+            'permanent_state' => $request->get('permanent_state'),
+            'permanent_country' => $request->get('permanent_country'),
+            'permanent_emergency_contact_name' => $request->get('permanent_emergency_contact_name'),
+            'permanent_emergency_contact_number' => $request->get('permanent_emergency_contact_number'),
+            'permanent_emergency_relationship' => $request->get('permanent_emergency_relationship'),
+
```
Comment (ithqindia) [inline app/Http/Controllers/AddressData.php:33]: empty line remove

File: app/Http/Controllers/RefreeData.php [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+
+use Illuminate\Http\Request;
+use App\Models\RefereeDetails;
+
+class RefreeData extends Controller
+{
+    public function store(Request $request)
+    {
+        $request->validate([
+
```
Comment (ithqindia) [inline app/Http/Controllers/RefreeData.php:13]: empty line remove

File: app/Http/Controllers/RefreeData.php [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+            'first_name' => 'required',
+            'last_name' => 'required',
+            'organisation' => 'required',
+            'position' => 'required',
+            'email_address' => 'required|email',
+
```
Comment (ithqindia) [inline app/Http/Controllers/RefreeData.php:19]: empty line remove

File: app/Http/Controllers/RefreeData.php [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+        ]);
+        RefereeDetails::create([
+            'first_name' => $request->get('first_name'),
+            'last_name' => $request->get('last_name'),
+            'organisation' => $request->get('organisation'),
+            'position' => $request->get('position'),
+            'email_address' => $request->get('email_address'),
+
```
Comment (ithqindia) [inline app/Http/Controllers/RefreeData.php:27]: empty line remove

File: app/Models/EmployeDetails.php [hunk 21]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class EmployeDetails extends Model
```
[the remark this answers is not in the document]
Comment (ithqindia) [reply app/Models/EmployeDetails.php:8]: Model name should be singular EmployeDetail not EmployeDetails

File: app/Models/EmployeAddress.php [hunk 22]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class EmployeAddress extends Model
+{
+    use HasFactory;
+    protected $table = "employe_address";
```
Comment (ithqindia) [inline app/Models/EmployeAddress.php:11]: table names should be plural employe_addresses

File: app/Models/RefereeDetails.php [hunk 23]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class RefereeDetails extends Model
```
[the remark this answers is not in the document]
Comment (ithqindia) [reply app/Models/RefereeDetails.php:8]: Model name should be singular

File: database/migrations/2021_12_11_062752_create_details_table.php [hunk 24]
```diff
@@ -0,0 +1,37 @@
...
+class CreateDetailsTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('employe_details', function (Blueprint $table) {
```
Comment (ithqindia) [inline database/migrations/2021_12_11_062752_create_details_table.php:16]: employee_details

File: database/migrations/2021_12_11_144809_create_addrescess_table.php [hunk 25]
```diff
@@ -0,0 +1,40 @@
...
+class CreateAddrescessTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('employe_address', function (Blueprint $table) {
```
Comment (ithqindia) [inline database/migrations/2021_12_11_144809_create_addrescess_table.php:16]: employee_addresses

**Review by ithqindia — APPROVED** [review]

**Review by ithqindia — COMMENTED** [review]
Change all controller names. Add Controller at end of name not at start

File: app/Models/EmployeAddress.php [hunk 26]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class EmployeAddress extends Model
```
Comment (ithqindia) [inline app/Models/EmployeAddress.php:8]: EmployeeAddress

File: app/Models/EmployeDetail.php [hunk 27]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class EmployeDetail extends Model
```
Comment (ithqindia) [inline app/Models/EmployeDetail.php:8]: EmployeeDetail

File: app/Models/LocalData.php [hunk 28]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class LocalData extends Model
```
Comment (ithqindia) [inline app/Models/LocalData.php:8]: LocalAddressData

File: app/Models/User.php [hunk 29]
```diff
@@ -41,4 +42,25 @@ class User extends Authenticatable
     protected $casts = [
         'email_verified_at' => 'datetime',
     ];
+    public function referee_data()
+    {
+        // This needs to be Model name
+        return $this->hasOne(RefereeDetail::class);
+    }
+
+    public function employe_data()
```
Comment (ithqindia) [inline app/Models/User.php:51]: employee_data

File: app/Models/User.php [hunk 30]
```diff
@@ -41,4 +42,25 @@ class User extends Authenticatable
...
+    {
+        // This needs to be Model name
+        return $this->hasOne(EmployeDetail::class);
+    }
+    public function local_data()
```
Comment (ithqindia) [inline app/Models/User.php:56]: local_address_data

File: app/Models/User.php [hunk 31]
```diff
@@ -41,4 +42,25 @@ class User extends Authenticatable
...
+    {
+        // This needs to be Model name
+        return $this->hasOne(LocalData::class);
+    }
+    public function address_data()
```
Comment (ithqindia) [inline app/Models/User.php:61]: employee_address_data

**Review by ithqindia — COMMENTED** [review]
Files names and class names needs to changed from Employe to Employee

File: resources/views/address-info.blade.php [hunk 32]
```diff
@@ -1,10 +1,10 @@
 <h1> permanent address data</h1>
 @if (isset($user->employee_address_data))
-    <h2>This is user informatio</h2>
+    <h2>User Informatio</h2>
```
Comment (ithqindia) [inline resources/views/address-info.blade.php:3]: User information

**Review by ithqindia — APPROVED** [review]

**Pull request closed without merging**